spatialmath/box.go

```diff
@@ -352,3 +355,113 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector
+func (b *box) ToPointCloud(options map[string]interface{}) []r3.Vector {
+	// check for user defined spacing
+	var faces [][]float64
+	var iter float64
+	if options["resolution"] != nil {
+		iter = options["resolution"].(float64)
+	} else {
+		iter = 0.15 // default spacing
+	}
+	// create points on box faces with box centered at (0, 0, 0)
+	for i := 0.0; i <= b.halfSize[0]; i += iter {
```

Comment:
should these for loops be made concurrent?

Reply:
Not to start with, but add a TODO pointing it out that that could be a place for a speedup if our ToPointcloud methods are too slow.